build/lib/module-copier.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+const copier = {};
+module.exports = copier;
+
+const fs = require('fs-extra');
+const path = require('path');
+
+const NODE_MODULES = 'node_modules';
+
+/**
+ * @param {string} projectPath absolute filepath for project root directory
+ * @param {string} targetPath absolute filepath for target directory to copy node_modules into
+ * @param {object} [options] options object
+ * @param {boolean} [options.includeOptional=true] whether to include optional dependencies when gathering
+ * @returns {Promise<void>} A Promise that resolves on completion
+ */
+copier.execute = async (projectPath, targetPath, options = { includeOptional: true }) => {
+	if (projectPath === null || projectPath === undefined) {
+		throw new Error('projectPath must be defined.');
+	}
+	if (targetPath === null || targetPath === undefined) {
+		throw new Error('targetPath must be defined.');
+	}
+
+	// resolve path names for file copying
+	projectPath = path.resolve(projectPath);
+	targetPath = path.resolve(targetPath);
+
+	// recursively gather the full set of dependencies/directories we need to copy
+	const root = new Dependency(null, 'fake-id', projectPath);
+	const directoriesToBeCopied = await root.getDirectoriesToCopy(options.includeOptional);
+
+	const dirSet = new Set(directoriesToBeCopied); // de-duplicate
+	// back to Array so we can #map()
+	const deDuplicated = Array.from(dirSet);
+
+	// Then copy them over
+	return Promise.all(deDuplicated.map(async directory => {
+		const relativePath = directory.substring(projectPath.length);
+		const destPath = path.join(targetPath, relativePath);
+		return fs.copy(directory, destPath, { overwrite: true }); // TODO: Allow incremental copying! Maybe use gulp/vinyl?
```

Comment:
It seems setting `dereference: true` on the opts here allows us to copy over a module that has been linked via `npm link`. That's the only difference I can see between the old method and this method (the old method would just completely ignore the link and reinstall everything).

Reply:
Before I think usage of `npm link` with our build process actually broke the build.

With this, the logic *looks* like it should copy the symlink. Setting to `dereference: true` would copy the dereferenced symlink. So I think we'd prefer to retain the symlink if we can for local development.